contrib/storage-hive/core/src/test/java/org/apache/drill/exec/hive/TestHiveStorage.java

```diff
@@ -36,6 +36,19 @@ public static void setupOptions() throws Exception {
     test(String.format("alter session set `%s` = true", PlannerSettings.ENABLE_DECIMAL_DATA_TYPE_KEY));
   }
 
+
+  @Test // DRILL-4083
+  public void testNativeScanWhenNoColumnIsRead() throws Exception {
+    try {
+      test(String.format("alter session set `%s` = true", ExecConstants.HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS));
+
+      String query = "SELECT count(*) FROM hive.readtest_parquet";
+      testPhysicalPlan(query, "hive-drill-native-parquet-scan");
+    } finally {
+      test(String.format("alter session set `%s` = false", ExecConstants.HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS));
```

Comment:
Will it make sense to change it back to default value in the finally block (in case we change the default value for this option in future release)?